www/source/javascripts/community.js

```diff
@@ -1,19 +1,128 @@
-var community_events = function(data) {
-    if ($(".community-events")) {
-        if (Array.isArray(data)) {
-            data.forEach(function(e) {
-                var ev = "<p>";
-                ev += e.title;
-                ev += "<br>";
-                ev += e.link;
-                ev += "<br>";
-                ev += e.pub_date;
-                ev += "<br>";
-                ev += e.description;
-                ev += "</p>";
-
-                $(".community-events").append(ev);
-            });
-        }
+$(function() {
+  var htmlDecode = function(input) {
+    var doc = new DOMParser().parseFromString(input, "text/html");
+    return doc.documentElement.textContent;
+  }
+
+  var makeEvent = function(e) {
+    if (e === undefined) {
```

Comment:
Do we sometimes see `undefined`s in the list of events? If so ... _scary_. (If not, we can probably remove this check, unless you mean to return `undefined` to the callers, which doesn't quite feel like what we want to do, either.)

If you're concerned about the integrity of the data, maybe a pass through it to validate it before simply bailing might be the way to go.